src/SeleniumLibrary/keywords/browsermanagement.py

```diff
@@ -123,23 +125,36 @@ def open_browser(self, url, browser='firefox', alias=None,
         Applying ``desired_capabilities`` argument also for local browser is
         new in SeleniumLibrary 3.1.
         """
+        def _resolve_browser_alias_to_browser_name(name):
+            return WebDriverCreator.browser_names.get(name.lower(), name)
+
         if is_truthy(remote_url):
             self.info("Opening browser '%s' to base url '%s' through "
                       "remote server at '%s'." % (browser, url, remote_url))
         else:
             self.info("Opening browser '%s' to base url '%s'." % (browser, url))
-        driver = self._make_driver(browser, desired_capabilities,
-                                   ff_profile_dir, remote_url)
-        driver = self._wrap_event_firing_webdriver(driver)
+
+        try:
+            driver, driver_index = self.ctx._get_driver(alias)
+            browser_name = _resolve_browser_alias_to_browser_name(browser)
+            if driver.name != browser_name:
+                self.warn("Multiple browsers with alias %s being used." %
+                          alias)
+                raise RuntimeError
```

Comment:
I was thinking that maybe a separate exception here could be useful to make it clear that `_get_driver()` can throw Runtime, Value & TypeExceptions if the look fails.